raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -663,11 +663,12 @@ public boolean isUnattached() {
     }
 
     public boolean isUnattachedNotVoted() {
-        return isUnattached() && !unattachedStateOrThrow().votedKey().isPresent();
+        EpochState fixedState = state;
+        return fixedState instanceof UnattachedState && !((UnattachedState) fixedState).votedKey().isPresent();
```

Comment:
Why the different implementation? One uses `maybeUnattachedState` while the this one reimplement `maybeUnattachedState`.